gnocchi/storage/redis.py

```diff
@@ -27,9 +27,36 @@ class RedisStorage(storage.StorageDriver):
     STORAGE_PREFIX = b"timeseries"
     FIELD_SEP = '_'
 
+    _SCRIPTS = {
+        "list_split_keys": """
+local metric_key = KEYS[1]
+local ids = {}
+local cursor = 0
+local substring = "([^" .. ARGV[2] .. "]*)"
+repeat
+    local result = redis.call("HSCAN", metric_key, cursor, "MATCH", ARGV[1])
+    cursor = tonumber(result[1])
+    for i, v in ipairs(result[2]) do
+        -- Only return keys, not values
+        if i % 2 ~= 0 then
```

Comment:
i've no idea what this is for but ok?

Reply:
HSCAN returns a table in format `{k1, v1, k2, v2, k3, v3…}` so using this condition allows to ignore `k1, k2, k3` and just push `v1, v2, v3` in `ids`

Reply:
you mean ignore v1, v2, v3...

Reply:
Sorry yeah, we want keys here, not values. :)

Reply:
cool cool. yeah i google'd most of the code. seems legit to someone that knows zero lua.